components/led_panel.py

```diff
@@ -0,0 +1,95 @@
+import wpilib
+
+
+class LEDPanel:
+    # Key: Match number, Value: Drop Bears match number (<=15)
+    # TODO: Fill with proper values when the draw comes out
+    matches: dict[int, int] = {1: 1}
+
+    def __init__(self) -> None:
+        # Custom packet: 00 000000
+        # match state, state specific information
+        self.match_state = 0
+        self.packet = 0b01000000
+        self.panel_port = wpilib.SerialPort(
+            baudRate=9600, port=wpilib.SerialPort.Port.kUSB1, dataBits=8
+        )
+        self.set_match_id()
+
+    def send_packet(self) -> None:
+        self.panel_port.write(self.packet.to_bytes())
+
+    def set_match_state(self) -> None:
+        self.packet &= 0b00111111
+
+        if wpilib.DriverStation.isEnabled():
+            # During
+            self.packet |= 0b01 << 6
+            if self.match_state != 1:
+                self.send_packet()
+                self.match_state = 1
+        elif wpilib.DriverStation.getMatchTime() <= 0:
+            # Post match
+            self.packet |= 0b10 << 6
+            if self.match_state != 2:
+                self.send_packet()
+                self.match_state = 2
+        else:
+            # Pre match
+            self.packet |= 0b00 << 6
+            if self.match_state != 0:
+                self.send_packet()
+                self.match_state = 0
+
+    def set_match_id(self) -> None:
+        match_type = wpilib.DriverStation.getMatchType()
+        match_id = 0
+        if (
+            match_type == wpilib.DriverStation.MatchType.kQualification
+            or match_type == wpilib.DriverStation.MatchType.kElimination
+        ):
+            match_number = wpilib.DriverStation.getMatchNumber()
+            if match_number in self.matches:
```

Comment:
This looks up the match number without using the match type. Is QF4 intended to be considered identical to Q4?